onnxruntime/core/providers/qnn/qnn_execution_provider.cc

```diff
@@ -496,7 +491,7 @@ QNNExecutionProvider::GetCapability(const onnxruntime::GraphViewer& graph_viewer
       if (partition && partition->sub_graph) {
         nodes_in_partition = partition->sub_graph->nodes.size();
 
-        if (nodes_in_partition == 1) {
+        if (nodes_in_partition == 1 && !is_qnn_ctx_model) {
```

Comment:
Perhaps this `!is_qnn_ctx_model` check is not necessary here. This if-statement marks the partition as invalid iff there is only one node inside the partition and that node is a Q/DQ op. Otherwise, nothing happens. This does not happen for a ctx model, correct?

Reply:
right, I should set the if (!is_qnn_ctx_model) outside the for loop to skip it.

---
In reply to: [1474836451](https://github.com/microsoft/onnxruntime/pull/18865#discussion_r1474836451) [](ancestors = 1474836451)

Reply:
Oh, now I remember, the purpose is to skip the following check for context cache model.

---
In reply to: [1474913612](https://github.com/microsoft/onnxruntime/pull/18865#discussion_r1474913612) [](ancestors = 1474913612,1474836451)

Reply:
The for loop is just counting the number of supported nodes and rejecting partitions with a single Q or DQ op. These should be fine to do even when using ctx model, right? I don't understand why we need to skip for ctx model.